src/game_inventory.cpp

```diff
@@ -1383,14 +1383,17 @@ void game_menus::inv::insert_items( avatar &you, item_location &holster )
         item &it = *holstered_item.first;
         bool success = false;
         if( !it.count_by_charges() ) {
-            if( all_pockets_rigid || holster.parents_can_contain_recursive( &it ) ) {
+            if( holster.get_item()->best_pocket( it, holster ).second != nullptr && ( all_pockets_rigid ||
```

Comment:
Does `can_contain` work here?

Reply:
it most likley would. i didn't because game_inventory::put_in() calls item::insert_item() which uses item_contents::find_pocket_for() which looks at all avaliable pockets of an item.

In short, this way we leave the option open to add a sheath with multiple pockets (like a bandolier for throwing knifes) can_contain() would ignore any additional pockets avaliable.

Reply:
Looking at the code `item::can_contain` and `item_contents::best_pocket` both check all pockets recursively whereas `item_contents::find_pocket_for` only checks top level pockets, so it might be better to just pass a flag to supress the debug message and show the failure message in `game_menus::inv::insert_items`.

Reply:
update: ok i switched to can_contain (weirdly enough best_pocket seems to be using can_contain anyways so that's fine)

I don't really wanna make it easier for people to ignore the debugmsg (it looks like put_in is used extensivly all over the project and i think the debugmsg is probably there to ensure that people catch problem cases before they reach this point)
Plus, using the neat ret_val pair is a bit moot point because it looks like the branch of the check that handles items with charges just straight up gets an int and doesn't follow the same standard.
Until someone changes that, i don't like the idea of one part handling error messages inside the method while the other gets it passed along (should be either or, not both imo)

Reply:
I've tried making a point against that debugmsg (as it is) before, but it wasn't too well received. (#44155)

Regarding your other point, as I understand it, it's somewhat of a goal to remove charges, so in a possible future the else case won't even exist anymore.

Reply:
We can probably add a new `ret_val` class that requires explicitly getting the failure message or suppressing it, and if a `ret_val` is not handled or suppressed, it prints a debug message when deconstructed.

Reply:
a uniform system of passing and handling errors would be lovly, but i think is beyond the scope of this minor bugfix.